saw-core/src/Verifier/SAW/SharedTerm.hs

```diff
@@ -1254,52 +1302,66 @@ scVectorReduced sc ety xs
 ------------------------------------------------------------
 -- Building terms using prelude functions
 
+-- | Create a term applying @Prelude.EqTrue@ to the given term.
 scEqTrue :: SharedContext -> Term -> IO Term
 scEqTrue sc t = scGlobalApply sc "Prelude.EqTrue" [t]
 
+-- | Create a @Prelude.Bool@-typed term from the given Boolean: @Prelude.True@
+-- for @True@, @Prelude.False@ for @False@.
 scBool :: SharedContext -> Bool -> IO Term
 scBool sc True  = scGlobalDef sc "Prelude.True"
 scBool sc False = scGlobalDef sc "Prelude.False"
 
+-- | Create a term representing the prelude Boolean type, @Prelude.Bool@.
 scBoolType :: SharedContext -> IO Term
 scBoolType sc = scGlobalDef sc "Prelude.Bool"
 
+-- | Create a term representing the prelude Natural type.
 scNatType :: SharedContext -> IO Term
 scNatType sc = scFlatTermF sc preludeNatType
 
+-- | Create a term representing a vector type, from a term giving the length
+-- and a term giving the element type.
```

Comment:
Per-argument docstrings would be nice (in general I like having them anywhere there are multiple arguments of the same type that mean different things)